spectrum-offchain-lm/src/bundle.rs

```diff
@@ -4,34 +4,345 @@ use async_trait::async_trait;
 use parking_lot::Mutex;
 
 use spectrum_offchain::data::unique_entity::{Confirmed, Predicted, Traced, Unconfirmed};
+use spectrum_offchain::data::OnChainEntity;
 
 use crate::data::bundle::StakingBundle;
 use crate::data::{AsBox, BundleId, BundleStateId, PoolId};
 
 pub mod process;
+pub mod rocksdb;
 
 #[async_trait]
 pub trait BundleRepo {
-    /// Select bundles corresponding to
+    fn get_epoch_length(&self) -> u32;
+    fn get_height_range(&self, epoch_ix: u32) -> EpochHeightRange;
```

Comment:
Those may be different for each `pool_id`, we store bundles for all pools in the network in this Repo.